### PR TITLE
Generation histogram of RGB colors

### LINKED ISSUE #7: Numpy
Given an image of choosing, create histogram of RGB colors found in the image using numpy.

### PR DESCRIPTION
**Pull request by RivkaZishinski**
## Description

This Pull Request adds a script that loads an image, computes the RGB histogram of the pixels in the image, and visualizes the histogram as a 3D scatter plot. The script uses the `NumPy` and `Matplotlib` libraries for numerical and graphical operations. The resulting plot shows the distribution of all colors in the image, with each pixel's RGB values represented as a point in the 3D space, and the color intensity represented as the point's size and color based on a plasma colormap.

Closes #7

### CODE REVIEW

**Comment by RivkaZishinski** [comment]
now i worked with the powershell and dont with the github workspace

**Comment by bdsoha** [comment]
> Is it good now?

The branches are great, please add a title and description

**Comment by RivkaZishinski** [comment]
Do add documentation to the code?

**Comment by bdsoha** [comment]
> Do add documentation to the code?

Documentation is good, but not always needed, and definitely not every line.
Please add a description to this PR and a readable title.

**Comment by RivkaZishinski** [comment]
**Description**

This Pull Request adds a script that loads an image, computes the RGB histogram of the pixels in the image, and visualizes the histogram as a 3D scatter plot. The script uses the NumPy and Matplotlib libraries for numerical and graphical operations. The resulting plot shows the distribution of all colors in the image, with each pixel's RGB values represented as a point in the 3D space, and the color intensity represented as the point's size and color based on a plasma colormap.

**Comment by bdsoha** [comment]
Why did you close this?

**Comment by RivkaZishinski** [comment]
sory , it was a mistake

**Comment by RivkaZishinski** [comment]
I finished the - a few clean up , what  more?

**Comment by RivkaZishinski** [comment]
when i  run the image from the command i dont get a error but it is not create a container , on the docker it is a button run(on the image) and it is create a container - what the problem?

**Comment by bdsoha** [comment]
> when i run the image from the command i dont get a error but it is not create a container , on the docker it is a button run(on the image) and it is create a container - what the problem?

I don't understand what you are referring to.

**Comment by bdsoha** [comment]
**Was your solution generated by ChatGPT?**

**Comment by RivkaZishinski** [comment]
I did use it a lot, as well as by searching on Google.

**Comment by bdsoha** [comment]
> I did use it a lot, as well as by searching on Google.

And the solution code iteself?

**Comment by RivkaZishinski** [comment]
by ChatGPT.

**Comment by bdsoha** [comment]
> by ChatGPT.

Without using ChatGPT, please re-write the code to only provide the *average* of *blue* pixels in the image.

**Comment by RivkaZishinski** [comment]
the code by chatJPT worked the best.

**Comment by bdsoha** [comment]
> the code by chatJPT worked the best.

That might be true in some cases *(and not true in many cases as well)*.
Again, you were assigned the task, not ChatGPT.
It was meant for you to learn and understand, and not just copy/paste.

**Comment by RivkaZishinski** [comment]
O.K.  I satrt

**Comment by RivkaZishinski** [comment]
it was good, Now I researched and I understand much more,does its good ?

File: numpy/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM python:latest
+RUN mkdir ~/.pip && echo "[global] \ntrusted-host = pypi.python.org pypi.org files.pythonhosted.org \n" > ~/.pip/pip.conf
+RUN pip install cv2
+RUN pip install numpy
+RUN pip install matplotlib
```
Comment (bdsoha) [inline numpy/Dockerfile:5]: Can this be inlined?
What happens every time a `RUN` command is executed? *(Hint: the image becomes a lot larger)*.

**Comment by RivkaZishinski** [comment]
do i need to write a code without cv2?

**Comment by bdsoha** [comment]
> do i need to write a code without cv2?

Explain your question, I didn't understand.

**Comment by RivkaZishinski** [comment]
this new code use with moudle - cv2.
do i need to write a new code that dont use with cv2

**Comment by bdsoha** [comment]
> this new code use with moudle - cv2. do i need to write a new code that dont use with cv2

No, you are free to use any package for this assignment.
In the future, I will explain best practices to choose open-source packages wisely.

My question was directed around the docker command `RUN`.
Every time the command is executed docker will add an additional layer to the image.
Images could become very large, very quickly when using too many `RUN`s.

You can obtain the same results by using:
```dockerfile
RUN pip install cv2 numpy matplotlib
```

**Comment by RivkaZishinski** [comment]
i got 
```
 => ERROR [3/6] RUN pip install cv2 numpy matplotlib 
```
i try to resolve this  - are you can help?

**Comment by bdsoha** [comment]
Did it work with multiple `RUN`s

**Comment by RivkaZishinski** [comment]
to install numpy  and  matplotlib  it work good but for cv2 it fall .
i try -pip install opencv-python and more things

**Comment by bdsoha** [comment]
> to install numpy and matplotlib it work good but for cv2 it fall . i try -pip install opencv-python and more things

Instead of having to deal with IT issues, try and find an alternate package.

**Comment by RivkaZishinski** [comment]
alternate package to cv2?

**Comment by bdsoha** [comment]
Yes

**Comment by RivkaZishinski** [comment]
now it is O.K.
i run also run it on the docker

File: numpy/RGBByNumpy.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+
+import numpy as np
```
Comment (bdsoha) [inline numpy/RGBByNumpy.py:2]:
```suggestion
import numpy as np

```

**Comment by bdsoha** [comment]
It is important to follow conventions when coding, it makes the overall repo look as if it was developed by a single person, when in fact there were multiple contributors.

Stick to `kebob-case` for file/folder names.
Stick to `snake_case` for python files.

File: numpy/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+## Getting Started
+
+To run the script, you need to have Docker installed on your machine. Then, follow these steps:
+
+   1. Clone this repository to your local machine.
+
+   2. Open a terminal window and navigate to the repository's directory.
+
+   3. Build the Docker image by running the following command:
```
Comment (bdsoha) [inline numpy/README.md:13]:
```suggestion
   1. Clone this repository to your local machine.
   2. Open a terminal window and navigate to the repository's directory.
   3. Build the Docker image by running the following command:

```

File: numpy/README.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+docker build -t rgb-by-numpy .
+```
+
+  4. Run the Docker container by running the following command:
+
+```
+docker run rgb-by-numpy
+
+```
+  5. The script will run and display a 3D plot of the RGB histogram of the image "light_blue.png" located in the "images" folder.
```
Comment (bdsoha) [inline numpy/README.md:25]:
```suggestion
  5. The script will run and display a 3D plot of the RGB histogram of the image `light_blue.png` located in the `images` folder.

```

File: numpy/README.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+## Customizing the Image
+
+If you want to visualize the RGB histogram of a different image, replace the file "light_blue.png" in the "images" folder with your own image file, and then rebuild and run the Docker image.
```
Comment (bdsoha) [inline numpy/README.md:29]:
```suggestion
If you want to visualize the RGB histogram of a different image, replace the file `light_blue.png` in the `images` folder with your own image file, and then rebuild and run the Docker image.

```

**Comment by RivkaZishinski** [comment]
what more?

File: numpy/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,7 @@
+FROM python:latest
+RUN mkdir ~/.pip && echo "[global] \ntrusted-host = pypi.python.org pypi.org files.pythonhosted.org \n" > ~/.pip/pip.conf RUN pip install cv2
+RUN pip install numpy matplotlib
+COPY RGBByNumpy.py /app/
```
Comment (bdsoha) [inline numpy/Dockerfile:4]:
```suggestion
COPY rgb-by-numpy.py /app/

```

File: numpy/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,7 @@
+FROM python:latest
+RUN mkdir ~/.pip && echo "[global] \ntrusted-host = pypi.python.org pypi.org files.pythonhosted.org \n" > ~/.pip/pip.conf RUN pip install cv2
+RUN pip install numpy matplotlib
+COPY RGBByNumpy.py /app/
+COPY assets/light_blue.png /app/images/
+WORKDIR /app
+CMD ["python", "RGBByNumpy.py"]
```
Comment (bdsoha) [inline numpy/Dockerfile:7]:
```suggestion
CMD ["python", "rgb_by_numpy.py"]

```

**Comment by bdsoha** [comment]
**🎉 Congrats on your first merged PR.**

You will soon be added as a collaborator to this repo.
In the meantime, please prepare a short presentation *(in English)* describing:

- The task in general.
- Things you learned along the way.
- Tips for other team-members for future tasks.

**Comment by bdsoha** [comment]
You have been sent an email invitation to the [SkyVar Organization](https://github.com/skyvar).
Please open your cooperate email and finish the registration process.

**Pull request merged**